apps/frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,57 @@
+import { IconButton } from "~/libs/components/components.js";
+import { useHandleClickOutside, useRef } from "~/libs/hooks/hooks.js";
+import { type ReactNode } from "~/libs/types/types.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	children: ReactNode;
+	isOpened: boolean;
+	onClose: () => void;
+	title: string;
+};
+
+const Modal = ({
+	children,
+	isOpened,
+	onClose,
+	title,
+}: Properties): JSX.Element | null => {
+	const dialogReference = useRef<HTMLDialogElement>(null);
+
+	useHandleClickOutside(dialogReference, onClose);
+
+	if (!isOpened) {
+		return null;
+	}
+
+	return (
+		<>
+			<div className={styles["modal-backdrop"]} />
+
+			<dialog
+				aria-label={title}
+				className={styles["modal-container"]}
+				ref={dialogReference}
+			>
+				<div className={styles["modal-content"]}>
+					<div className={styles["modal-header-title"]}>
+						<h3>{title}</h3>
+						<div className={styles["modal-close"]}>
+							<IconButton
+								iconHeight={10.61}
+								iconName="cross"
+								iconWidth={10.61}
```

Comment:
We were told that it is better to use round numbers, in this case 10 or 11

Reply:
Also, in Figma all icons come with a frame, which makes them equal in size and similar to other icons in the app. We should download icons with this frame.
![Screenshot 2024-08-26 at 12 03 18](https://github.com/user-attachments/assets/8900b1d7-1521-4f28-a412-b3a31df9f89e)